src/app.js

```diff
@@ -1,13 +1,186 @@
+      }
+      let $tripList = $('#trip-list');
+      $tripList.empty();
+      if (filteredTrips && filteredTrips.length > 0) {
+        filteredTrips.forEach((trip) => {
+          $tripList.append(allTripsTemplate(trip.attributes));
+        });
+      } else {
+        $tripList.append('<h1>No Trips Fit The Criteria =(</h1>');
+      }
+    }
+  },
+  fetchTrip() {
+    const trip = tripList.get($(this).data('id'));
+    trip.fetch({
+      success: events.successfulTripFetch,
+      error: events.failedTripFetch,
+    });
+  },
+  successfulTripFetch(trip) {
+    let $tripDetails = $('#trip-details');
+    $tripDetails.empty();
+    $tripDetails.append(tripDetailsTemplate(trip.attributes));
+  },
+  failedTripFetch(response) {
+    events.addStatusMessagesFromHash('#page-status-messages', 'error', response.responseJSON.errors);
+  },
+  hideModal(){
+    $('.status-title').empty();
+    $('.status-messages section ul').empty();
+    $('.status-messages').hide();
+    $.modal.close();
+  },
+  addTrip(event){
+    event.preventDefault();
+    const tripData = {};
+    tripFields.forEach( (field) => {
+      const val = $(`#create-trip-form input[name=${field}]`).val();
+      if (val != '') tripData[field] = val;
+    });
+    const trip = new Trip(tripData);
+    if (trip.isValid()) {
+      trip.save({}, {
+        success: events.successfulSaveTrip,
+        error: events.failSaveTrip
+      });
+    } else { // save is invalid
+      events.addStatusMessagesFromHash('#modal-status-messages', 'errors', trip.validationError);
+    }
+  },
+  successfulSaveTrip(trip, response){
+    $('#create-trip-form .input').val('');
+    events.hideModal();
+    tripList.add(trip);
+    events.addStatusMessagesFromHash('#page-status-messages', 'success', {message: 'Trip has been successfully added'});
+  },
+  failSaveTrip(trip, response){
+    events.addStatusMessagesFromHash('#modal-status-messages', 'errors', response.responseJSON.errors);
+    trip.destroy();
+  },
+  addStatusMessagesFromHash(jquerySelector, statusTitle, collection){
+    let tripAttributes = collection;
+    tripAttributes['status'] = statusTitle;
+    $('.status-title').empty();
+    $(jquerySelector).empty();
+    $(jquerySelector).append(statusMessageTemplate({trip: tripAttributes}));
+    if (statusTitle === 'success'){
+      $(jquerySelector).show().delay(3000).fadeOut();
+    } else {
+      $(jquerySelector).show();
+    }
+  },
+  addReservation(event){
+    event.preventDefault();
+    const reservationData = {};
+    reservationFields.forEach( (field) => {
+      const val = $(`#create-reservation-form input[name=${field}]`).val();
+      if (val != '') reservationData[field] = val;
+    });
+    const reservation = new Reservation(reservationData);
+    if (reservation.isValid()) {
+      const tripID = $(this).data('id');
+      reservation.urlRoot = `${(new Trip()).urlRoot}${tripID}/reservations`;
```

Comment:
Another way to achieve this result would be to have the `Reservation` model take a property (_not_ an attribute) for the associated `Trip`, and then define `urlRoot` as a function which calculates the correct URL based on that trip:

```javascript
// in models/reservation.js
const Reservation = Backbone.Model.extend({
  initialize(attributes, options) {
    this.trip = options.trip;
  },
  urlRoot() {
    return `${this.trip.url()}/reservations`;
  },
  // rest of your model code
});
```

```javascript
// in app.js
const trip = tripList.findWhere({ id: tripID });
const reservation = new Reservation(reservationData, { trip: trip });
```